audio-worklet/noisegate-test.html

```diff
@@ -0,0 +1,527 @@
+<!--
+Copyright 2017 Google Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<!DOCTYPE html>
+  <head>
+    <script src="test/testharness.js"></script>
+    <script src="test/testharnessreport.js"></script>
+    <script src="test/audit-util.js"></script>
+    <script src="test/audit.js"></script>
+    <script src="src/noisegate-script-processor.js"></script>
+    <script id="noisegate-test-code">
+
+      // Sample rate is a power of two to avoid interpolation from delay node.
+      // It is held artificially low for easy inspection of audio buffers.
+      const sampleRate = 4096;
+      const testDurationSec = 1;
+      const testFrames = testDurationSec * sampleRate;
+      const bufferSize = 512;
+      const scriptProcessorLatency = bufferSize * 2;
+
+      // An appropriate default setting for alpha depends on the frequency
+      // spectrum of the signal and its amplitude relative to the threshold.
+      // When the signal is below the threshold, the gate should be closed. But
+      // with lower frequency signals, the gate will not stay fully closed if it
+      // is very near the threshold. For example, if the oscillator frequency is
+      // set to 40 and alpha is set to 0.90, then the gate won't remain fully
+      // closed at an amplitude greater than 82% of the threshold. 
+      // A lower alpha leads to a lower envelope latency and a smaller
+      // difference between when the gate is expected to be fully closed and
+      // when it actually is fully closed. When frequency is 440Hz and the
+      // signal gain is 90% of the threshold, the alpha setting with shortest 
+      // delay is 0.64 to 0.68 but with an alpha of 0.63 the noise gate will 
```

Comment:
There are two considerations. The first is the delay and the second is suppressing high frequency oscillation. What I am trying to say is that the alpha setting with shortest delay that  mutes enough high frequency oscillation to avoid hysteris is 0.64.